### ISSUE TITLE
Misaligned cursor when editor.fontLigatures enabled

### ISSUE THREAD

**Issue by Ziao** [Microsoft/vscode#19915]
Not a biggie, but it may at least help people running into the same issue;
TLDR; Quick fix is to disable ``` editor.fontLigatures```.

- VSCode Version: Code 1.9.0 (27240e71ef390bf2d66307e677c2a333cebf75af, 2017-02-02T08:23:30.856Z)
- OS Version: Darwin x64 16.3.0
- Extensions:
(confirmed not dependent on extensions)

Since one of the recent updates (2 days ago or so), when ``` editor.fontLigatures``` is enabled, and certain letter combinations are replaced by a ligature, the cursor will become out of sync, that is to say, the actual cursor is several characters behind. This affects only the selected line. Please see screenshots for clarification.

Notice where my cursor is (note: "fi" in "files" is replaced with a ligature, and takes up the space for one character)
![before](https://cloud.githubusercontent.com/assets/1612079/22617070/e2b5ba46-eabb-11e6-888c-cc98eca740e2.png)

After I try to enter a dash to separate the words, it ends up here.
![after](https://cloud.githubusercontent.com/assets/1612079/22617069/e2b27282-eabb-11e6-9eb8-126044cec615.png)

---

Steps to Reproduce:

1. set ``` editor.fontLigatures``` to ```true```
2. make sure your ```editor.fontFamily``` has ```Menlo``` first, for example's sake.
3. Create a new (javascript) file, and enter ```let files = []; //this is a comment``` into it. You'll notice very clearly that as soon as the ligatures kick in, the cursor will be out of sync with the actual location.

Again, not a game-breaking bug, but it's a bug nonetheless.
Other than that, I'm very satisfied with vscode. Keep it up :)

**Comment by alexandrudima** [comment]
@Ziao thank you for reporting. This is a regression, caused by the new monospace optimization work. You can workaround by using `"editor.disableMonospaceOptimizations": true` in your settings.

**Comment by meden** [comment]
Hello,

I understand that there is a workaround for this issue, but the issue still stands, so maybe you could consider reopening it: the editor is broken (for me) with its default configuration.

Moreover, the setting `editor.disableMonospaceOptimizations` is marked as `Unknown configuration setting` by the editor, making it even more obscure.

Thank you.

**Comment by alexandrudima** [comment]
@meden AFAIK this issue is resolved, are you encountering something similar. If yes, then please create a new issue, with information such as: your vs code version, your operating system, and your settings. A screenshot of the bug would also help.
